staging/test/Makefile

```diff
@@ -0,0 +1 @@
+include $(RIOTBASE)/Makefile.base
```

Comment:
I would prefer showcasing a real module that is used by the application and has source code + headers.
Also `test` is a pretty common name so would prefer something saying `showcase of staging module`.

I would be great if it could include a `Makefile.include` and a `Makefile.dep` to show how they must be written.
`Makefile.dep` must be protected as it is always parsed even if the module is not enabled and it may not be clear for everybody.

Reply:
> I would prefer showcasing a real module that is used by the application and has source code + headers.
> Also `test` is a pretty common name so would prefer something saying `showcase of staging module`.

- renamed "test" to "showcase"
- filled with some life (include folder with header, a C file containing a field)
- changed "hello-staging" to use the module

Reply:
> I would be great if it could include a `Makefile.include` and a `Makefile.dep` to show how they must be written.

It has a Makefile.include now.


> `Makefile.dep` must be protected as it is always parsed even if the module is not enabled and it may not be clear for everybody.

Hm, good point. Actually I prefer not to parse ```Makefile.{dep, include}``` of broken modules at all. I've changed staging.mk accordingly. What do you think?

Reply:
I would even only define `STAGING_MODULES_USED` with the non broken ones. No reason to have 3 variables here.